src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,232 @@
+  selectAllText: string
+  title: string
+}
+
+export interface Option {
+  id: string
+  text: string
+  iconUrl?: string
+}
+
+function MultiSelectBottomSheet<T extends string>({
+  forwardedRef,
+  onClose,
+  onOpen,
+  options,
+  setSelectedOptions,
+  selectedOptions,
+  selectAllText,
+  title,
+}: MultiSelectBottomSheetProps<T>) {
+  const { t } = useTranslation()
+  const scrollViewRef = useRef<ScrollView>(null)
+
+  const isEveryOptionSelected = options.length === selectedOptions.length
+
+  const handleClose = () => {
+    onClose?.()
+  }
+
+  return (
+    <BottomSheetBase
+      forwardedRef={forwardedRef}
+      onClose={handleClose}
+      onOpen={onOpen}
+      backgroundStyle={{ backgroundColor: 'transparent' }}
+      handleIndicatorStyle={{ width: 0 }}
+    >
+      <BottomSheetScrollView
+        forwardedRef={scrollViewRef}
+        testId={'MultiSelectBottomSheet'}
+        containerStyle={styles.bottomSheetScrollView}
+      >
+        <View style={[styles.option, styles.borderRadiusTop]}>
+          <Text style={styles.boldTextStyle}>{title}</Text>
+        </View>
+        <View style={styles.optionsContainer}>
+          <ScrollView style={{ height: OPTION_HEIGHT * MAX_OPTIONS_IN_VIEW }}>
+            <Option
+              text={selectAllText}
+              isSelected={isEveryOptionSelected}
+              onPress={() => setSelectedOptions(options)}
+            />
+            {options.map((option) => (
+              <Option
+                text={option.text}
+                iconUrl={option.iconUrl}
+                isSelected={
+                  !!selectedOptions.find((selectedOption) => selectedOption.id === option.id) &&
+                  !isEveryOptionSelected
+                }
+                onPress={() => {
+                  if (isEveryOptionSelected) {
+                    setSelectedOptions([option])
+                  } else {
+                    if (selectedOptions.find((selectedOption) => selectedOption.id === option.id)) {
+                      setSelectedOptions(
+                        selectedOptions.filter((selectedOption) => selectedOption.id !== option.id)
+                      )
+                    } else {
+                      setSelectedOptions([...selectedOptions, option])
+                    }
+                  }
+                }}
+              />
+            ))}
+          </ScrollView>
+        </View>
+        <View style={styles.doneButtonContainer}>
+          <Touchable
+            testID="MultiSelectBottomSheet/Done"
+            style={styles.doneButton}
+            onPress={handleClose}
+          >
+            <Text style={styles.boldTextStyle}>{t('done')}</Text>
+          </Touchable>
+        </View>
+      </BottomSheetScrollView>
+    </BottomSheetBase>
+  )
+}
+
+interface OptionProps {
+  onPress?: () => void
+  text: string
+  isSelected: boolean
+  iconUrl?: string
+}
+function Option({ onPress, text, iconUrl, isSelected }: OptionProps) {
```

Comment:
I guess only the interface Option is exported, but there can be potential collisions of this also needs to be exported.